PWGHF/Core/HfMlResponseD0ToKPi.h

```diff
@@ -0,0 +1,264 @@
+#FEATURE, static_cast < uint8_t>(InputFeaturesD0ToKPi::FEATURE) \
+  }
+
+// Check if the index of mCachedIndices (index associated to a FEATURE)
+// matches the entry in EnumInputFeatures associated to this FEATURE
+// if so, the inputFeatures vector is filled with the FEATURE's value
+// by calling the corresponding GETTER from OBJECT
+#define CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)   \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::FEATURE): { \
+    inputFeatures.emplace_back(OBJECT.GETTER());              \
+    break;                                                    \
+  }
+
+// Specific case of CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)
+// where OBJECT is named candidate and FEATURE = GETTER
+#define CHECK_AND_FILL_VEC_D0(GETTER)                        \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::GETTER): { \
+    inputFeatures.emplace_back(candidate.GETTER());          \
+    break;                                                   \
+  }
+
+// Variation of CHECK_AND_FILL_VEC_D0_FULL(OBJECT, FEATURE, GETTER)
+// where GETTER is a method of hfHelper
+#define CHECK_AND_FILL_VEC_D0_HFHELPER(OBJECT, FEATURE, GETTER) \
+  case static_cast<uint8_t>(InputFeaturesD0ToKPi::FEATURE): {   \
+    inputFeatures.emplace_back(hfHelper.GETTER(OBJECT));        \
+    break;                                                      \
+  }
+
+namespace o2::analysis
+{
+enum class InputFeaturesD0ToKPi : uint8_t {
+  collisionId = 0,
+  posX,
+  posY,
+  posZ,
+  xSecondaryVertex,
+  ySecondaryVertex,
+  zSecondaryVertex,
+  errorDecayLength,
+  errorDecayLengthXY,
+  chi2PCA,
+  rSecondaryVertex,
+  decayLength,
+  decayLengthXY,
+  decayLengthNormalised,
+  decayLengthXYNormalised,
+  impactParameterNormalised0,
+  ptProng0,
+  pt2Prong0,
+  ptProng1,
+  pt2Prong1,
+  pxProng0,
+  pyProng0,
+  pzProng0,
+  pxProng1,
+  pyProng1,
+  pzProng1,
+  impactParameter0,
+  impactParameter1,
+  errorImpactParameter0,
+  errorImpactParameter1,
+  nSigTpcPi0,
+  nSigTpcKa0,
+  nSigTofPi0,
+  nSigTofKa0,
+  nSigTpcPi1,
+  nSigTpcKa1,
+  nSigTofPi1,
+  nSigTofKa1,
+  m,
+  maxNormalisedDeltaIP,
+  impactParameterProduct,
+  cosThetaStar,
+  pt,
+  p,
+  cpa,
+  cpaXY,
+  ct,
+  eta,
+  phi,
+  y,
+  e
+};
+
+template <typename TypeOutputScore = float>
+class HfMlResponseD0ToKPi : public HfMlResponse<TypeOutputScore>
+{
+ public:
+  /// Default constructor
+  HfMlResponseD0ToKPi() = default;
+  /// Default destructor
+  virtual ~HfMlResponseD0ToKPi() = default;
+
+  HfHelper hfHelper;
+
+  /// Method to get the input features vector needed for ML inference
+  /// \param candidate is the Dplus candidate
+  /// \param prong0 is the candidate's prong0
+  /// \param prong1 is the candidate's prong1
+  /// \return inputFeatures vector
+  template <typename T1, typename T2>
+  std::vector<float> getInputFeatures(T1 const& candidate,
+                                      T2 const& prong0, T2 const& prong1)
+  {
```

Comment:
Differently from the Dplus, in the case of the D0 we should handle the two possible mass hypotheses (KPi or PiK). This means that we should have the flexibility to fill the vector of variables with one hypothesis or the other for all the variables that depend on this mass hypothesis, such as the cost*, see https://github.com/AliceO2Group/O2Physics/blob/1204d8f12614b5601381c70138bcdee4224ccf5b/PWGHF/Core/HfHelper.h#L73-L83
 My proposal would be to add an argument here that specifies whether the vector should be filled for the hypothesis `D0toPiK` (first daughter as a pion, second one as a kaon) or the hypothesis `D0bartoKPi` (first daughter as a kaon, second one as a pion)